native/packages/react-native-bpk-component-banner-alert/src/AnimateAndFade.js

```diff
@@ -16,85 +16,133 @@
  * limitations under the License.
  */
 
-import { Animated, ViewPropTypes } from 'react-native';
-import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import BpkAnimateHeight from 'react-native-bpk-component-animate-height';
+import React, { Component } from 'react';
+import { View, Animated, ViewPropTypes } from 'react-native';
 import { animationDurationSm } from 'bpk-tokens/tokens/base.react.native';
 
-const FADE_DURATION = animationDurationSm;
-const ANIMATE_HEIGHT_DURATION = animationDurationSm;
+const COLLAPSED_HEIGHT = 0.01;
 
 class AnimateAndFade extends Component {
   constructor(props) {
     super(props);
 
-    this.state = {
-      visible: !this.props.animateOnEnter,
-      opacity: new Animated.Value(0.01),
-    };
+    const isCollapsed = this.props.animateOnEnter || !this.props.show;
 
-    this.animate = this.animate.bind(this);
+    this.shouldRenderHeight = true;
+    this.innerViewRef = null;
+
+    this.height = isCollapsed ? new Animated.Value(COLLAPSED_HEIGHT) : null;
+    this.opacity = new Animated.Value(isCollapsed ? 0 : 1);
   }
 
   componentDidMount() {
-    requestAnimationFrame(() => {
-      this.animate(true);
-    });
+    const captureHeightAndAnimate = (x, y, width, height) => {
+      if (!this.height) {
+        this.height = new Animated.Value(height);
+      }
+
+      this.animate(x, y, width, height);
+    };
+
+    this.measure(captureHeightAndAnimate);
   }
 
   componentWillReceiveProps(nextProps) {
-    if (nextProps.show !== this.state.visible) {
-      this.animate(!this.state.visible);
+    this.shouldRenderHeight = true;
+
+    if (nextProps.show === this.props.show && this.props.show) {
+      this.shouldRenderHeight = false;
     }
   }
 
-  animate(show) {
-    Animated.timing(this.state.opacity, {
-      toValue: show ? 1 : 0,
-      duration:
-        (show && this.props.animateOnEnter) ||
-        (!show && this.props.animateOnLeave)
-          ? FADE_DURATION
-          : 0,
-      delay: show ? ANIMATE_HEIGHT_DURATION : 0,
```

Comment:
this was confusing - the previous animation was based on delays and durations between opacity and height.

No it's just a sequence (see line 84).